### PR TITLE
[core] [Label] render label element instead of div

### LINKED ISSUE #1971: Label JS API outputs a `div` instead of a `label`
### Bug report

- __Package version(s)__: 1.35.0
- __Browser and OS versions__: Electron, Mac OSX 10.13.2

#### Steps to reproduce

1. Import the `Label` component
1. Paste the example from the documentation in the render function
1. See how a `div` is outputted instead of a `label`

#### Actual behavior

A `div` is returned from the render method of the `Label` component

#### Expected behavior

A `label` should be returned from the render method of the `Label` component

Note: I checked and the style `.pt-label` selectors in the core `scss` are specific for the `label` element, so they do not apply to a `div`.

### PR DESCRIPTION
**Pull request by gianmarcotoso**
Fixes #1971

#### Changes proposed in this pull request:

The `Label` component originally returned a `div` element, which was not supported by the `.pt-label` selector. I modified the `label.tsx` file so that it returns a `label` element instead of a `div`.

#### Reviewers should focus on:

I have written some tests for the `Label` component (which were previously missing), but they are very naïve. Maybe there's more stuff that could be tested.

### CODE REVIEW

**Comment by palantirtech** [comment]
Thanks for your interest in [palantir/blueprint](https://github.com/palantir/blueprint), @gianmarcotoso! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

**Comment by gianmarcotoso** [comment]
It's failing on the `DateRangePicker`, which I haven't even touched (and it's not part of the core). Is this failing on the master branch as well?

File: packages/core/test/forms/labelTests.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+ * Licensed under the terms of the LICENSE file distributed with this project.
+ */
+
+import { assert } from "chai";
+import { shallow } from "enzyme";
+import * as React from "react";
+
+import { Classes, Label } from "../../src/index";
+
+describe("<Label>", () => {
```
Comment (cmslewis) [inline packages/core/test/forms/labelTests.tsx]: @adidahiya FYI these tests

**Comment by giladgray** [comment]
@gianmarcotoso what's the status of this PR? first step for you is to merge latest `develop`, which _should_ resolve build issues. please close this PR if you don't have time to finish it and i'll pick up this feature (encountered this issue in my own app development 😄 )

**Comment by gianmarcotoso** [comment]
Sorry I totally forgot! I'll do it tomorrow so that hopefully we'll get
this thing done! :)

**Comment by giladgray** [comment]
@gianmarcotoso push it up! let's see if it fails on circle. those two tests are flaky, particularly when run in chrome (local) vs phantom (circle).

**Comment by gianmarcotoso** [comment]
Nope, still borked - even though only one is failing now, still on the `Overlay`.

**Comment by adidahiya** [comment]
Not your fault on the broken unit test -- I think it's broken on develop.

Regarding the unit tests -- yes, they are too simple. Please remove them, I don't think it's worth keeping around tautological tests that just verify the basic structure of the component.

**Comment by gianmarcotoso** [comment]
Ok, I'll remove the tests, merge develop again for good measure and push again later today (hopefully!)

**Comment by giladgray** [comment]
@gianmarcotoso still waiting on that push...

**Comment by gianmarcotoso** [comment]
Sorry, pushed right now!

**Comment by adidahiya** [comment]
thanks @gianmarcotoso!

**Comment by gianmarcotoso** [comment]
You're welcome, thank you for Blueprint! :)

**Pull request merged**